scripts/layerops.py

```diff
@@ -0,0 +1,721 @@
+# Example Use:
+# from objects.layer import Layer
+
+# demo = Layer()
+# demo.load_file("C:\Users\attack\Downloads\layer.json")
+# demo2 = Layer()
+# demo2.load_input(Existing_Layer_String_Previously_Loaded)
+
+# lo = layerops([demo, demo2])
+# lo.process(score=lambda x: x[0] * x[1], out_Name="example 1", out_Desc="This is an list example")
+# out_layer = lo.get_processed()
+# out_layer.export_file("C:\demo_layer1.json")
+
+# lo2 = layerops({'a': demo, 'b': demo2})
+# lo2.process(score=lambda x: x['a'], color=lambda x: x['b'], out_Desc="This is a dict example")
+# lo2.edit_meta(name="example 2")
+# out_layer2 = lo2.get_processed()
+# out_layer2.export_file("C:\demo_layer2.json")
+
+import copy
+from objects.layer import Layer
+
+class InvalidFormat(Exception):
+    pass
+
+class BadLambda(Exception):
+    pass
+
+class MismatchedDomain(Exception):
+    pass
+
+class layerops:
+    """
+        Initialization - takes a list or dict of Layer objects, and prepares to process them
+
+        :param data: A dict or list of Layer objects.
+        :raises InvalidFormat: An error indicating that a the layer data wasn't provided in a list or dict
+    """
+    def __init__(self, data):
+        self.data = data
+        if isinstance(data, dict):
+            temp = {}
+            for entry in data.keys():
+                temp[entry] = data[entry]._get('techniques')
+            self.mode = 'dict'
+            self._merge_to_template(key=list(data.keys())[0])
+        elif isinstance(data, list):
+            temp = []
+            for entry in data:
+                temp.append(entry._get('techniques'))
+            self.mode = 'list'
+            self._merge_to_template()
+        else:
+            raise InvalidFormat
+        self.da = temp
+        self.corpus = self._build_template(temp)
+
+        # Default values for fields if they aren't supplied
+        self.default_values = {
+            "comment": "",
+            "enabled": True,
+            "color": "#ffffff",
+            "score": 1,
+            "metadata": []
+        }
+        self.processed = None
+
+    """
+        Processes the ingested data using User provided Lambdas
+        :param score: lambda to calculate score
+        :param comment: lambda to generate comments
+        :param enabled: lambda to determine enabled status
+        :param metadata: lambda to generate metadata
+        :param defaults: dictionary containing desired default values for missing data element values
+        :param out_Name: new name to apply to the resulting layer
+        :param out_Desc: new description to apply to the resulting layer
```

Comment:
These two parameters should be lambda using as well. Users who are doing for example score multiplication may want to define the name as literally "layerA * layerB" with layerA and layerB substituting with layer names. If they want a static name for the output layer they should just define a static return value for the lambda function, just like they would for the rest of the params.

Reply:
Can do.